queryparser/sparql/src/main/java/org/eclipse/rdf4j/query/parser/sparql/TupleExprBuilder.java

```diff
@@ -1495,12 +1481,10 @@ private TupleExpr createTupleExprForNegatedPropertySet(NegatedPropertySet nps, i
 
 		TupleExpr patternMatch = null;
 
-		// build a regular statement pattern (or a join of several patterns if
-		// the
-		// object list has more than
-		// one item)
+		// build a regular statement pattern (or a join of several patterns if the object list has more than one item)
 		if (filterCondition != null) {
-			for (ValueExpr objVar : nps.getObjectList()) {
+			for (ValueExpr obj : nps.getObjectList()) {
+				final Var objVar = mapValueExprToVar(obj);
```

Comment:
There's a bit of formatting cleanup in this file, but this line is the actual fix.